cookbooks/aws-parallelcluster-config/resources/mount_fsx.rb

```diff
@@ -105,3 +105,59 @@
     end
   end
 end
+
+action :unmount do
+  fsx_fs_id_array = new_resource.fsx_fs_id_array.dup
+  fsx_fs_type_array = new_resource.fsx_fs_type_array.dup
+  fsx_shared_dir_array = new_resource.fsx_shared_dir_array.dup
+  fsx_dns_name_array = new_resource.fsx_dns_name_array.dup
+  fsx_mount_name_array = new_resource.fsx_mount_name_array.dup
+  fsx_volume_junction_path_array = new_resource.fsx_volume_junction_path_array.dup
+
+  fsx_fs_id_array.each_with_index do |fsx_fs_id, index|
+    fsx_shared_dir = fsx_shared_dir_array[index]
+    fsx_dns_name = fsx_dns_name_array[index]
+    fsx_fs_type = fsx_fs_type_array[index]
+    fsx_volume_junction_path = fsx_volume_junction_path_array[index]
```

Comment:
I think that, not in this PR and probably neither in this feature, we should go from having arrays of properties for the same object to having simple arrays of objects. 

So moving from something like this:

```
items_id_array.each_with_index do |item_id, index|
    attribute_1 = attribute_1_array[index]
    ...
    attribute_n = attribute_n_array[index]
end
```

to something like this:

```
items.each do |item|
    attribute_1 = item.attribute_1
    ...
    attribute_n = item.attribute_n
end
```

Reply:
fsx_shared_dir_array, fsx_dns_name  comes from dns.json from the user_data. I think this problem comes from that dna.json are passing attribute array as string to save some space, instead of having a map to store each of them. Maybe we are doing that because the user data has limited size. Yes in the future we may be able to improve it.